src/main/scala/rise/core/traverse.scala

```diff
@@ -196,23 +183,23 @@ object traverse {
     override def `type`[T <: Type] : T => M[T] = return_
   }
 
-  case class Pure[T](unwrap : T)
-  implicit object PureMonad extends Monad[Pure] {
-    override def return_[T] : T => Pure[T] = t => Pure(t)
-    override def bind[T,S] : Pure[T] => (T => Pure[S]) => Pure[S] =
-      v => f => v match { case Pure(v) => f(v) }
-  }
-
-  implicit object OptionMonad extends Monad[Option] {
-    def return_[T]: T => Option[T] = Some(_)
-    def bind[T, S]: Option[T] => (T => Option[S]) => Option[S] = v => v.flatMap
-  }
-
-  trait PureTraversal extends Traversal[Pure] { override def monad = PureMonad }
+  trait PureTraversal extends Traversal[Pure] {override def monad : PureMonad.type = PureMonad }
   trait PureExprTraversal extends PureTraversal with ExprTraversal[Pure]
+  trait PairMonoidTraversal[F,M[_]] extends Traversal[InMonad[M]#SetFst[F]#Type] {
```

Comment:
Can we name this `StatefulTraversal` with `fstMonoid` being renamed to `state`?

Or maybe `AccumulatorTraversal` with `fstMonoid` -> `accumulator`?

Reply:
Could we then also default the Monad to `Pure` (as done for the `PureTraversal`)?

Reply:
Done, thank you!